packages/circuit-ui/components/NotificationToast/NotificationToast.tsx

```diff
@@ -0,0 +1,223 @@
+};
+
+type NotificationToastWrapperProps = HTMLAttributes<HTMLDivElement> & {
+  variant: Variant;
+};
+
+const toastWrapperStyles = ({
+  theme,
+  variant,
+}: NotificationToastWrapperProps & StyleProps) => css`
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  position: relative;
+  background-color: ${theme.colors.bodyBg};
+  padding: ${theme.spacings.kilo} ${theme.spacings.mega};
+  border-radius: ${theme.borderRadius.byte};
+  border: ${theme.borderWidth.mega} solid ${theme.colors[colorMap[variant]]};
+  overflow: hidden;
+  transition: opacity 200ms ease-in-out, height 200ms ease-in-out,
+    visibility 200ms ease-in-out;
+`;
+
+const NotificationToastWrapper = styled('div')<NotificationToastWrapperProps>(
+  toastWrapperStyles,
+);
+
+const contentStyles = ({ theme }: StyleProps) => css`
+  display: flex;
+  flex-direction: column;
+  align-items: flex-start;
+  padding-right: ${theme.spacings.peta};
+  padding-left: ${theme.spacings.mega};
+`;
+
+const Content = styled('div')(contentStyles);
+
+const StyledIcon = styled.span(
+  ({ theme, variant }: StyleProps & { variant: Variant }) =>
+    css`
+      display: block;
+      align-self: flex-start;
+      flex-grow: 0;
+      flex-shrink: 0;
+      line-height: 0;
+      color: ${theme.colors[colorMap[variant]]};
+    `,
+);
+
+const closeButtonStyles = ({ theme }: StyleProps) => css`
+  flex-grow: 0;
+  flex-shrink: 0;
+  align-self: flex-start;
+  margin-top: -${theme.spacings.bit};
+  margin-bottom: -${theme.spacings.bit};
+  margin-left: auto;
+`;
+
+const StyledCloseButton = styled(CloseButton)(closeButtonStyles);
+
+export function NotificationToast({
+  variant = 'info',
+  body,
+  headline,
+  onClose,
+  closeButtonLabel,
+  isVisible,
+  tracking,
+  duration, // this is the auto-dismiss duration, not the animation duration. We shouldn't pass it to the wrapper along with ...props
+  ...props
+}: NotificationToastProps): JSX.Element {
+  const contentElement = useRef(null);
+  const [isOpen, setOpen] = useState(isVisible);
+  const [height, setHeight] = useState(getHeight(contentElement));
+  const [, setAnimating] = useAnimation();
+  useEffect(() => {
+    setAnimating({
+      duration: 200,
+      onStart: () => {
+        setHeight(getHeight(contentElement));
+        // Delaying the state update until the next animation frame ensures that
+        // the browsers renders the new height before the animation starts.
+        window.requestAnimationFrame(() => {
+          setOpen(isVisible);
+        });
+      },
+      onEnd: () => {
+        setHeight(DEFAULT_HEIGHT);
+      },
+    });
+  }, [isVisible, setAnimating]);
+
+  return (
+    <NotificationToastWrapper
+      ref={contentElement}
+      style={{
+        opacity: isOpen ? 1 : 0,
+        height: isOpen ? height : 0,
+        visibility: isOpen ? 'visible' : 'hidden',
+      }}
+      variant={variant}
+      {...props}
+    >
+      <StyledIcon as={iconMap[variant]} variant={variant} />
```

Comment:
We should have an optional (hidden) label for the button.

If the toast body is self-explanatory, the icon doesn't need a label, e.g.:

```
✔️ Saved successfully.
```

This will speak `Saved successfully`—we don't need it to say `Success, saved successfully`

But imagine another scenario where the icon _does_ add contextual information.

After clicking the "log out" button:

```
✔️ You were logged out.
```

When the session expires:

```
⚠️ You were logged out.
```

Although these two have the same body, they should speak respectively `Success, you were logged out.` and `Warning, you were logged out`.

Reply:
We added the iconLabel prop, with the hideVisually, to cover both scenarios.